### PR TITLE
Fix and check links with hyperlink

### PR DESCRIPTION
**Pull request by ggrossetie**
#### `lint:links`

This task uses [`hyperlink`](https://www.npmjs.com/package/hyperlink) to detect invalid and "inefficient" (i.e., links with unnecessary redirections) links.

#### `.github/workflows/docs.yml`

This workflow will build the documentation and run the task `lint:links`.

#### `.github/workflows/notify.yml`

This workflow will trigger a build at https://github.com/neo4j-documentation/docs-refresh as soon as some documentation changed and is pushed on the `4.0` branch.

### CODE REVIEW

File: .github/workflows/notify.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  trigger_publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Trigger Developer Event
+        uses: peter-evans/repository-dispatch@master
+        with:
+          token: ${{ secrets.BUILD_ACCESS_TOKEN }}
```
Comment (ggrossetie) [inline .github/workflows/notify.yml:18]: @jexp @adam-cowley Could you please make sure that the token `BUILD_ACCESS_TOKEN` is available on this repository? Thanks!

**Comment by ggrossetie** [comment]
You can take a look at the result on my fork: https://github.com/Mogztter/neo4j-spark-connector/runs/1645285005?check_suite_focus=true

**Comment by conker84** [comment]
@adam-cowley can you please review it?

File: .github/workflows/notify.yml (hunk 1 above)
Comment (moxious) [reply .github/workflows/notify.yml:18]: @adam-cowley ping ☝️

**Comment by moxious** [comment]
@mroiter-larus to review document, plug in build token and merge / take action

**Comment by mroiter-larus** [comment]
@moxious The BUILD_ACCESS_TOKEN secret has been created on the neo4j-contrib repo. Now the PR could be merged

**Pull request merged**